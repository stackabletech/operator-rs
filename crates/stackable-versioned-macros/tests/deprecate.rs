use stackable_versioned_macros::versioned;

#[test]
fn deprecate() {
    #[versioned(
        version(name = "v1alpha1"),
        version(name = "v1beta1"),
        version(name = "v1")
    )]
    struct Foo {
        #[versioned(deprecated(since = "v1beta1", note = "gone"))]
        deprecated_bar: usize,
        baz: bool,
    }
}
