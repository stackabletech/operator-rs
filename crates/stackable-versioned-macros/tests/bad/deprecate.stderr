error: deprecation must be done using #[versioned(deprecated(since = "VERSION"))]
  --> tests/bad/deprecate.rs:10:9
   |
10 |         #[deprecated]
   |         ^
